modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/SelectTracksWorkflowOperationHandler.java

```diff
@@ -0,0 +1,500 @@
+    // course, but with more than two, we might want something better. Hopefully, this will be easy to improve later
+    // on.
+
+    // First case: We have only tracks with non-hidden video streams. So we keep them all and possibly cut away audio.
+    if (allNonHidden(augmentedTracks, SubTrack.VIDEO)) {
+      final Optional<AudioMuxing> audioMuxing = getConfiguration(workflowInstance, CONFIG_AUDIO_MUXING)
+              .map(AudioMuxing::fromConfigurationString);
+      // For both special options, we need to find out if we have exactly one audio track present
+      final Optional<AugmentedTrack> singleAudioTrackOpt = findSingleAudioTrack(augmentedTracks);
+      final boolean multipleVideo = augmentedTracks.size() > 1;
+      if (multipleVideo && audioMuxing.map(m -> m == AudioMuxing.DUPLICATE).orElse(Boolean.FALSE) && singleAudioTrackOpt
+              .isPresent()) {
+        // Special option: If we have multiple video tracks, but only one audio track: copy this audio track to
+        // all video tracks.
+        final AugmentedTrack singleAudioTrack = singleAudioTrackOpt.get();
+        for (final AugmentedTrack t : augmentedTracks) {
+          if (t.track != singleAudioTrack.track) {
+            final TrackJobResult jobResult = mux(t.track, singleAudioTrack.track, mediaPackage);
+            result.add(jobResult);
+          }
+        }
+      } else if (multipleVideo && audioMuxing.map(m -> m == AudioMuxing.FORCE).orElse(Boolean.FALSE)
+              && singleAudioTrackOpt.isPresent()) {
+        // Special option: if the only audio track we have selected is not in the video track of "force-target", we
+        // copy it there (and remove the original audio track).
+        final AugmentedTrack singleAudioTrack = singleAudioTrackOpt.get();
+        final String forceTargetOpt = getConfiguration(workflowInstance, CONFIG_FORCE_TARGET)
+                .orElse(FORCE_TARGET_DEFAULT);
+
+        final Optional<AugmentedTrack> forceTargetTrackOpt = findTrackByFlavorType(augmentedTracks, forceTargetOpt);
+
+        if (!forceTargetTrackOpt.isPresent()) {
+          throw new IllegalStateException(
+                  String.format("\"%s\" set to \"%s\", but target flavor \"%s\" not found!",
+                          CONFIG_AUDIO_MUXING,
+                          AudioMuxing.FORCE.value, forceTargetOpt));
+        }
+
+        final AugmentedTrack forceTargetTrack = forceTargetTrackOpt.get();
+
+        if (singleAudioTrack.track != forceTargetTrack.track) {
+          // Copy it over...
+          final TrackJobResult muxResult = mux(forceTargetTrack.track, singleAudioTrack.track, mediaPackage);
+          result.add(muxResult);
+
+          // ...and remove the original
+          final TrackJobResult hideAudioResult = hideAudio(singleAudioTrack.track, mediaPackage);
+          result.add(hideAudioResult);
+        }
```

Comment:
You need to copy tracks you do not mux as well. Otherwise, the output of this operation is impossible to predict.

Example:

- You have two video files `presenter/source` and `presentation/source`. The first has video only, the second has video and audio.
- You set this operation to force audio in `presentation` and set the target flavor to `*/selected`
- The operation will run, claim success but since the audio was already in the right place, there is no track with the flavor `*/selected`
- Result: The next operation will likely fail.

here is a simple patch which fixes the problem. I'm not sure if I covered all edge cases:

```diff
diff --git a/modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/SelectTracksWorkflowOperationHandler.java b/modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/SelectTracksWorkflowOperationHandler.java
index dd135ddbd..385687bdd 100644
--- a/modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/SelectTracksWorkflowOperationHandler.java
+++ b/modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/SelectTracksWorkflowOperationHandler.java
@@ -39,11 +39,13 @@ import org.opencastproject.workflow.api.WorkflowOperationResult;
 import org.opencastproject.workflow.api.WorkflowOperationTagUtil;
 import org.opencastproject.workspace.api.Workspace;
 
+import org.apache.commons.io.FilenameUtils;
 import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
+import java.net.URI;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
@@ -51,6 +53,7 @@ import java.util.List;
 import java.util.Optional;
 import java.util.SortedMap;
 import java.util.TreeMap;
+import java.util.UUID;
 import java.util.function.Consumer;
 import java.util.stream.Collectors;
 
@@ -273,6 +276,8 @@ public class SelectTracksWorkflowOperationHandler extends AbstractWorkflowOperat
           if (t.track != singleAudioTrack.track) {
             final TrackJobResult jobResult = mux(t.track, singleAudioTrack.track, mediaPackage);
             result.add(jobResult);
+          } else {
+            result.add(copyTrack(t.track));
           }
         }
       } else if (multipleVideo && audioMuxing.map(m -> m == AudioMuxing.FORCE).orElse(Boolean.FALSE)
@@ -302,6 +307,15 @@ public class SelectTracksWorkflowOperationHandler extends AbstractWorkflowOperat
           // ...and remove the original
           final TrackJobResult hideAudioResult = hideAudio(singleAudioTrack.track, mediaPackage);
           result.add(hideAudioResult);
+        } else {
+          result.add(copyTrack(singleAudioTrack.track));
+        }
+
+        // Just copy the rest of the tracks to ensure they got the correct output flavor
+        for (final AugmentedTrack augmentedTrack : augmentedTracks) {
+          if (augmentedTrack.track != singleAudioTrack.track && augmentedTrack.track != forceTargetTrack.track) {
+            result.add(copyTrack(augmentedTrack.track));
+          }
         }
       } else {
         // No special options selected, or conditions for special options don't match.
@@ -497,4 +511,27 @@ public class SelectTracksWorkflowOperationHandler extends AbstractWorkflowOperat
       return new AugmentedTrack(t, hideAudio, hideVideo);
     }).collect(Collectors.toList());
   }
+
+  private TrackJobResult copyTrack(Track track) throws WorkflowOperationException {
+    Track copiedTrack = (Track) track.clone();
+    copiedTrack.setIdentifier(UUID.randomUUID().toString());
+    try {
+      // Generate a new filename
+      String targetFilename = copiedTrack.getIdentifier();
+      String extension = FilenameUtils.getExtension(track.getURI().getPath());
+      if (extension.length() > 0) {
+        targetFilename += "." + extension;
+      }
+
+      // Copy the files on dis and put them into the working file repository
+      logger.debug("Start copying element {}.", track.getURI());
+      URI newUri = workspace.put(track.getMediaPackage().getIdentifier().toString(), copiedTrack.getIdentifier(),
+              targetFilename, workspace.read(track.getURI()));
+      copiedTrack.setURI(newUri);
+    } catch (IOException | NotFoundException e) {
+      throw new WorkflowOperationException(String.format("Error while copying track %s", track.getIdentifier()), e);
+    }
+
+    return new TrackJobResult(copiedTrack, 0);
+  }
 }
```